test/facade/sdkFacade.test.ts

```diff
@@ -340,56 +346,180 @@ describe('A SDK facade', () => {
 
         sdkFacade.anonymize();
 
-        expect(tracker.unsetToken).toBeCalledTimes(1);
+        expect(context.setToken).toHaveBeenLastCalledWith(null);
     });
 
-    test('should not allow setting an invalid token', () => {
+    test('should allow to unset a token', () => {
+        const context = createContextMock();
+
+        jest.spyOn(Sdk, 'init')
+            .mockImplementationOnce(config => {
+                const sdk = Sdk.init(config);
+
+                jest.spyOn(sdk, 'context', 'get').mockReturnValue(context);
+
+                return sdk;
+            });
+
         const sdkFacade = SdkFacade.init({
             appId: appId,
             track: false,
         });
 
-        function setToken(): void {
-            sdkFacade.setToken(null as unknown as string);
-        }
+        sdkFacade.unsetToken();
+
+        expect(context.setToken).toBeCalledTimes(0);
+
+        sdkFacade.setToken(Token.issue(appId, 'c4r0l'));
+
+        sdkFacade.unsetToken();
 
-        expect(setToken).toThrow(Error);
-        expect(setToken).toThrow('The token must be of type string.');
+        expect(context.setToken).toHaveBeenLastCalledWith(null);
     });
 
-    test('should allow to unset a token', () => {
-        const tracker = jest.genMockFromModule<Tracker>('../../src/tracker');
-        tracker.setToken = jest.fn();
-        tracker.unsetToken = jest.fn();
+    test('should allow to set a token', () => {
+        const context = createContextMock();
+
+        jest.spyOn(Sdk, 'init')
+            .mockImplementationOnce(config => {
+                const sdk = Sdk.init(config);
+
+                jest.spyOn(sdk, 'context', 'get').mockReturnValue(context);
+
+                return sdk;
+            });
+
+        const sdkFacade = SdkFacade.init({
+            appId: appId,
+            track: false,
+        });
+
+        const carolToken = Token.issue(appId, 'c4r0l');
+
+        sdkFacade.setToken(carolToken);
+
+        expect(context.setToken).toBeCalledWith(carolToken);
+        expect(context.setToken).toBeCalledTimes(1);
+    });
+
+    test('should provide the current token', () => {
+        const context = createContextMock();
 
         jest.spyOn(Sdk, 'init')
             .mockImplementationOnce(config => {
                 const sdk = Sdk.init(config);
 
+                jest.spyOn(sdk, 'context', 'get').mockReturnValue(context);
+
+                return sdk;
+            });
+
+        const sdkFacade = SdkFacade.init({
+            appId: appId,
+            track: false,
+        });
+
+        const carolToken = Token.issue(appId, 'c4r0l');
+
+        expect(sdkFacade.getToken()).toBeNull();
+
+        sdkFacade.setToken(carolToken);
+
+        expect(sdkFacade.getToken()).toEqual(carolToken);
+    });
+
+    test('should allow to refresh the token of the current user', () => {
```

Comment:
Here

Reply:
This tests refreshing a identified token, sending only one `userSignedIn`. We are missing a test refreshing anonymous token, that should not send any tracked event.

Reply:
oh, got it